File: app/src/main/java/me/varlez/movies/MoviesApp.java
package me.varlez.movies;

import android.app.Application;

import me.varlez.movies.common.di.component.DaggerMoviesComponent;
import me.varlez.movies.common.di.component.MoviesComponent;
import me.varlez.movies.common.di.module.MoviesModule;

/**
 * Custom application class handling Dagger graph creation.
 */
public class MoviesApp extends Application {

    protected MoviesComponent moviesComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initAppComponent();
    }

    protected void initAppComponent() {
        this.moviesComponent = DaggerMoviesComponent
                .builder()
                .moviesModule(new MoviesModule(this))
                .build();
    }

    public MoviesComponent getMoviesComponent() {
        return this.moviesComponent;
    }

}

File: app/src/main/java/me/varlez/movies/MoviesApp.kt
package me.varlez.movies

import android.app.Application
import android.content.Context
import me.varlez.movies.common.di.component.AppComponent
import me.varlez.movies.common.di.component.DaggerAppComponent
import me.varlez.movies.common.di.module.NetworkModule

/**
 * Custom application class handling Dagger graph creation.
 */
open class MoviesApp : Application() {

    lateinit var appComponent: AppComponent
        protected set

    override fun onCreate() {
        super.onCreate()
        initAppComponent()
    }

    protected open fun initAppComponent() {
        this.appComponent = DaggerAppComponent
                .builder()
                .networkModule(NetworkModule(this))
                .build()
    }

    companion object {
        fun get(context: Context) = context.applicationContext as MoviesApp
    }

}
